singularity-eos/eos/eos.hpp

```diff
@@ -88,6 +122,144 @@ class IdealGas {
                               Real &bmod, Real &dpde, Real &dvdt,
                               Real *lambda = nullptr) const;
 
+  /*
+  Vector versions of the member functions are only available for CPU
+  
+  RealIndexer must have an operator[](int) that returns a Real. e.g., Real*
+  ConstRealIndexer is as RealIndexer, but assumed const type.
+  LambdaIndexer must have an operator[](int) that returns a Real*. e.g., Real**
+  */
+  template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
```

Comment:
In principle, `rhos` and `sies` could be different `Indexer` types.

Reply:
What circumstances do you think would lead to input indexers having different types? Assuming a collection of states, I'd assume that the input variables would necessarily use the same indexing scheme to travel over the collection of states.

Reply:
Yeah you're probably right, so we can leave it for now. I was thinking of some scenario where rho and sie were sparsely populated in different ways somehow.

Reply:
Let's revisit if we find a circumstance where this occurs though. It will be fairly straightforward to change